pysat/_instrument.py

```diff
@@ -985,7 +985,8 @@ def _load_data(self, date=None, fid=None):
             raise TypeError('Metadata returned must be a pysat.Meta object')
 
         # let user know if data was returned or not
-        if len(data) > 0:
+        ind = data.index if self.pandas_format else data.indexes
+        if len(ind) > 0:
```

Comment:
same.

I really want to keep checks on data type to the absolute minimum. Folks, including us, shouldn't have to know ahead of time what kind of data is coming down.

Reply:
This is the bit that has a data object that becomes attached to the pysat.instrument.  The data object (if xarray) does not have an empty property and won't until after this routine is run.